### PR TITLE
404 Screen

### PR DESCRIPTION
**Pull request by shrek-7**
- route to 404 screen if route doesn't match

Signed-off-by: shreyas <shreyas.srivastava199@gmail.com>

### CODE REVIEW

File: client/src/index.js [hunk 1]
```diff
@@ -37,7 +38,9 @@ const store = customStore.create();
 ReactDOM.render(
   <div id='next-root'>
     <Provider store={store}>
-      <App routes={routes}/>
+      <BrowserRouter>
```
Comment (pgobin) [inline client/src/index.js]: Only one `BrowserRouter` should exist. This nests `<BrowserRouter>` in `App.js` within it. https://github.com/ReactTraining/react-router/issues/5291.

File: client/src/state/app/AppActions.js [hunk 2]
```diff
@@ -37,6 +37,8 @@ const { Types, Creators } = createActions({
 
   refreshOnNextSocketReceive:        ['flag'],
   feedReceive:                       ['payload'],
+
```
Comment (pgobin) [inline client/src/state/app/AppActions.js]: Routing should be divorced from application state.

File: client/src/containers/app/AppHelper.js [hunk 3]
```diff
@@ -112,6 +113,8 @@ export const appDispatch = (dispatch) => {
     startRefresh:      ()    => dispatch(AppActions.refreshBegin()),
     stopRefresh:       ()    => dispatch(AppActions.refreshEnd()),
     forceSocketError:  ()    => dispatch(AppActions.socketMaxAttemptsReached()),
+    pageNotFound:      (status)    =>
```
Comment (pgobin) [inline client/src/containers/app/AppHelper.js]: Routing should be divorced from application state.

File: client/src/containers/app/App.js [hunk 4]
```diff
@@ -262,10 +313,15 @@ class App extends Component {
...
             <Route exact path='/signup' component={Signup}/>
             { !isAuthenticated && <Redirect to='/login'/> }
+            { !isPageNotFound && <Redirect to='/404-page-not-found'/> }
             <Route
               exact
               path='/browse'
               render={() => <Browse {...this.props}/>}/>
+            <Route
+              exact
+              path='/404-page-not-found'
```
Comment (pgobin) [inline client/src/containers/app/App.js]: 404 redirect shouldn't change the URL.

File: client/src/components/layouts/PageNotFound.css [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  font-size: 12rem;
+  font-weight: bold;
+  margin: 0;
+}
+
+.error-layout__text {
+  font-size: 2rem;
+  text-align: center;
+  margin: 0;
+}
```
Comment (pgobin) [inline client/src/components/layouts/PageNotFound.css]: - All files should have a copyright header.
- All files should have an EOF newline.
- Optionally, CSS rules should be sorted alphabetically.
- Prefer HEX to RGB.
- Class and ID names should be hyphen-delimited.

I'll see what I can do for automating EOF + CSS sorting/name checks.

**Comment by shrek-7** [comment]
PR is updated with suggested changes.

**Comment by shrek-7** [comment]
version update has been added.

**Pull request merged**